tests/scripts/thread-cert/Cert_5_3_07_DuplicateAddress.py

```diff
@@ -31,67 +31,77 @@
 import unittest
 
 import node
+import command
+import config
+import mle
 
-LEADER = 1
+DUT_LEADER = 1
 ROUTER1 = 2
 ROUTER2 = 3
-ED1 = 4
-ED2 = 5
-ED3 = 6
+MED1 = 4
+SED1 = 5
+MED3 = 6
 
-MTDS = [ED1, ED2, ED3]
+MTDS = [MED1, SED1, MED3]
 
 class Cert_5_3_7_DuplicateAddress(unittest.TestCase):
     def setUp(self):
         self.nodes = {}
         for i in range(1,7):
             self.nodes[i] = node.Node(i, (i in MTDS))
 
-        self.nodes[LEADER].set_panid(0xface)
-        self.nodes[LEADER].set_mode('rsdn')
-        self.nodes[LEADER].add_whitelist(self.nodes[ROUTER1].get_addr64())
-        self.nodes[LEADER].add_whitelist(self.nodes[ROUTER2].get_addr64())
-        self.nodes[LEADER].add_whitelist(self.nodes[ED3].get_addr64())
-        self.nodes[LEADER].enable_whitelist()
+        self.nodes[DUT_LEADER].set_panid(0xface)
+        self.nodes[DUT_LEADER].set_mode('rsdn')
+        self.nodes[DUT_LEADER].add_whitelist(self.nodes[ROUTER1].get_addr64())
+        self.nodes[DUT_LEADER].add_whitelist(self.nodes[ROUTER2].get_addr64())
+        self.nodes[DUT_LEADER].add_whitelist(self.nodes[MED3].get_addr64())
+        self.nodes[DUT_LEADER].enable_whitelist()
 
         self.nodes[ROUTER1].set_panid(0xface)
         self.nodes[ROUTER1].set_mode('rsdn')
-        self.nodes[ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64())
-        self.nodes[ROUTER1].add_whitelist(self.nodes[ED1].get_addr64())
+        self.nodes[ROUTER1].add_whitelist(self.nodes[DUT_LEADER].get_addr64())
+        self.nodes[ROUTER1].add_whitelist(self.nodes[MED1].get_addr64())
         self.nodes[ROUTER1].enable_whitelist()
         self.nodes[ROUTER1].set_router_selection_jitter(1)
 
         self.nodes[ROUTER2].set_panid(0xface)
         self.nodes[ROUTER2].set_mode('rsdn')
-        self.nodes[ROUTER2].add_whitelist(self.nodes[LEADER].get_addr64())
-        self.nodes[ROUTER2].add_whitelist(self.nodes[ED2].get_addr64())
+        self.nodes[ROUTER2].add_whitelist(self.nodes[DUT_LEADER].get_addr64())
+        self.nodes[ROUTER2].add_whitelist(self.nodes[SED1].get_addr64())
         self.nodes[ROUTER2].enable_whitelist()
         self.nodes[ROUTER2].set_router_selection_jitter(1)
 
-        self.nodes[ED1].set_panid(0xface)
-        self.nodes[ED1].set_mode('rsn')
-        self.nodes[ED1].add_whitelist(self.nodes[ROUTER1].get_addr64())
-        self.nodes[ED1].enable_whitelist()
+        self.nodes[MED1].set_panid(0xface)
+        self.nodes[MED1].set_mode('rsn')
+        self.nodes[MED1].add_whitelist(self.nodes[ROUTER1].get_addr64())
+        self.nodes[MED1].enable_whitelist()
 
-        self.nodes[ED2].set_panid(0xface)
-        self.nodes[ED2].set_mode('rsn')
-        self.nodes[ED2].add_whitelist(self.nodes[ROUTER2].get_addr64())
-        self.nodes[ED2].enable_whitelist()
+        self.nodes[SED1].set_panid(0xface)
+        self.nodes[SED1].set_mode('rsn')
```

Comment:
remove 'r' for SED at least.
In fact, for all scenarios involving SED in TESTPLAN, it is designed that SED should only retrieve only stable network data. That is necessary especially for scenarios whose aim is to test network data. So I would prefer to set the mode of SED to be 's' at least from now on.

Reply:
Are you going to update the mode for SED1?